projects/schematics/src/shared/utils/lib-utils.ts

```diff
@@ -1047,3 +1026,95 @@ export function runExternalSpartacusLibrary(
 function createModuleFileName(config: FeatureConfig): string {
   return `${dasherize(config.moduleName)}-feature.module.ts`;
 }
+
+/**
+ * Used to sort the features in the correct order.
+ */
+function calculateSort(libraryA: string, libraryB: string): number {
+  const indexA = installationOrder.indexOf(libraryA);
+  const indexB = installationOrder.indexOf(libraryB);
+
+  /**
+   * In case a feature module is _not_ found in the `installationOrder`,
+   * we want to sort it at the end of the list.
+   */
+  return (indexA > -1 ? indexA : Infinity) - (indexB > -1 ? indexB : Infinity);
+}
+
+export function orderInstalledFeatures<T extends LibraryOptions>(
+  options: T
+): Rule {
+  return (tree: Tree, context: SchematicContext): void => {
+    context.logger.info('Ordering the installed Spartacus features...');
+
+    const basePath = process.cwd();
+    const { buildPaths } = getProjectTsConfigPaths(tree, options.project);
+    for (const tsconfigPath of buildPaths) {
+      const { appSourceFiles } = createProgram(tree, basePath, tsconfigPath);
+
+      let spartacusFeaturesModule: SourceFile | undefined;
+      for (const sourceFile of appSourceFiles) {
+        if (
+          sourceFile
+            .getFilePath()
+            .includes(`${SPARTACUS_FEATURES_MODULE}.module.ts`)
+        ) {
+          spartacusFeaturesModule = sourceFile;
+          break;
+        }
+      }
+
+      if (!spartacusFeaturesModule) {
+        continue;
+      }
+
+      const collectedModules = collectInstalledModules(spartacusFeaturesModule);
+      if (!collectedModules) {
+        continue;
+      }
+
+      if (options.debug) {
+        context.logger.info(
+          `Sorting the installed Spartacus features according to the dependency graph: ${installationOrder.join(
+            ', '
+          )}`
+        );
+      }
+      const spartacusCoreModules = collectedModules.spartacusCoreModules.map(
+        (spartacusCoreModule) => spartacusCoreModule.getText()
+      );
+      const featureModules = collectedModules.featureModules
+        .sort((moduleA, moduleB) =>
+          calculateSort(moduleA.spartacusLibrary, moduleB.spartacusLibrary)
+        )
+        .map((featureModule) => featureModule.moduleNode.getText());
+      const unrecognizedModules = collectedModules.unrecognizedModules.map(
+        (unrecognizedModule) => unrecognizedModule.getText()
+      );
+
+      const moduleImportsProperty = getModulePropertyInitializer(
+        spartacusFeaturesModule,
+        'imports',
+        false
+      );
+      if (!moduleImportsProperty) {
+        continue;
+      }
+
+      if (collectedModules.warnings.length) {
+        context.logger.warn(
+          'The following modules were not recognized due to various reasons:'
+        );
+        for (const warning of collectedModules.warnings) {
+          context.logger.warn(warning);
+        }
+      }
+
+      const orderedModules: string[] = spartacusCoreModules
```

Comment:
coolio, I see how it's done now :d 

Took a closer look today under the hood 👍